src/components/navigation/TheCanvasTools.vue

```diff
@@ -118,18 +134,13 @@ export default class TheCanvasTools extends Vue {
     }
   }
 
-  zoomOutFunction (): void {
-    this.zoomOut()
-    EventBus.$emit('zoom')
+  zoomFunction (zoomOut: boolean): void {
+    (zoomOut) ? this.zoomOut() : this.zoomIn()
   }
 
-  zoomInFunction (): void {
-    this.zoomIn()
-    EventBus.$emit('zoom')
```

Comment:
No need for emits here. All we need to do is update the state and the StageWatchers will handle the rest :)